src/components/InputMedia/InputMedia.tsx

```diff
@@ -0,0 +1,96 @@
+import { useRef, useState, useEffect } from 'react';
+
+import classNames from 'classnames';
+
+import scss from './InputMedia.module.scss';
+
+import emptyInputGrey from './assets/imageEmptyGray.svg';
+
+import { IInputMediaProps } from './InputMedia.types';
+
+function InputMedia(props: IInputMediaProps) {
+  const fileInputRef = useRef<HTMLInputElement | null>(null);
+
+  const [mediaType, setMediaType] = useState('');
+  const [mediaSelected, setMediaSelected] = useState<string[]>([]);
+
+  useEffect(() => {
+    if (props.files && props.files.length > 0) {
+      const mediaSelectedUrls = props.files.map((file) =>
+        URL.createObjectURL(file)
+      );
+
+      setMediaSelected(mediaSelectedUrls);
+      const mediaType = props.files[0].type.match('(image|video)');
+      setMediaType(mediaType ? mediaType[0] : '');
+    }
+  }, [props.files]);
```

Comment:
Me explica pq isso é necessário. Se esse é o input media, pq ele recebe media do pai? É pq da lista? Se for isso, tudo bem, mas porque precisa de um `useEffect`?